File: app/src/main/java/com/buddha/mindboard/data/datasource/RemoteDataSource.java
package com.buddha.mindboard.data.datasource;


import com.buddha.mindboard.data.api.ApiService;
import com.buddha.mindboard.data.datasource.base.BaseDataSource;
import com.buddha.mindboard.data.datasource.base.DataSource;

import io.reactivex.Observable;

/**
 * Created by Buddha Saikia on 06-10-2018.
 */

public class RemoteDataSource extends BaseDataSource implements DataSource.Greetings {

    private ApiService apiService;

    public RemoteDataSource(ApiService apiService) {
        this.apiService = apiService;
    }

    @Override
    public Observable<String> greetings() {
        return Observable.just("Greetings from API sample")
                .compose(this.<String>applySchedulersIO());
    }
}

File: app/src/main/java/com/buddha/mindboard/data/datasource/RemoteDataSource.kt
package com.buddha.mindboard.data.datasource


import com.buddha.mindboard.data.api.ApiService
import com.buddha.mindboard.data.datasource.base.BaseDataSource
import com.buddha.mindboard.data.datasource.base.DataSource

import com.buddha.mindboard.data.model.Datum
import io.reactivex.Observable
import retrofit2.Response

/**
 * Created by Buddha Saikia on 13-11-2018.
 */

class RemoteDataSource(private val apiService: ApiService) : BaseDataSource(), DataSource {

    override fun getData(): Observable<Response<List<Datum>>>? {
        return apiService.data.compose(applySchedulersIO())
    }
}
